### PR TITLE
Filter dropdown

### PR DESCRIPTION
**Pull request by kojodoesdesign**
**High level description:**
Adds a "Created By" dropdown filter in the Broker Dashboard that allows the user to filter the table. 

**Technical details:**
- `CreatedByContainer` makes a call to the `list_all_submissions` endpoint, removes duplicate names, and passes the remaining values to populate the dropdown of an [Awesomeplete](https://www.npmjs.com/package/awesomplete) Typeahead component.
- `DropdownTypeahead` gets the value typed into the [Awesomeplete](https://www.npmjs.com/package/awesomplete) Typeahead component, and created an unchecked checkbox. 
- Upon Clicking the Checkbox, the staged `CreatedBy` filter tags are processed for display
- Processes applied `CreatedBy` filters and adds them to the `list_submissions` `filters` request param

**Link to JIRA Ticket:**
[DEV-1366](https://federal-spending-transparency.atlassian.net/browse/DEV-1366)

The following are ALL required for the PR to be merged:
- [x] Frontend review completed
- [x] Design review completed

### CODE REVIEW

File: src/js/components/SharedComponents/DropdownTypeahead.jsx [hunk 1]
```diff
@@ -0,0 +1,350 @@
...
+  onDropdownChange() {
+    const currentState = this.state.dropdownopen;
+    this.setState({
+      dropdownopen: !currentState,
+    });
+  }
+
+  getSelectedName() {
+    if (this.state.unselectedValues.includes(this.state.value) || this.state.selectedValues.includes(this.state.value)) {
+      alert('This name is already used, please use the checkbox in the dropdown.');
```
Comment (ebdabbs) [inline src/js/components/SharedComponents/DropdownTypeahead.jsx]: We should check with the UX team, but there is probably a better solution than using an alert here.

File: src/js/components/SharedComponents/DropdownTypeahead.jsx [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+  changedText(e) {
+    this.setState({
+      value: e.target.value,
+    }, this.detectEmptySuggestions);
+  }
+
+  detectEmptySuggestions() {
+    if (this.state.value.length > 2 && this.typeahead.suggestions.length === 0) {
+      if (!this.state.showWarning) {
+        // we need to show a warning that no matching agencies were found
```
Comment (ebdabbs) [inline src/js/components/SharedComponents/DropdownTypeahead.jsx]: I think this comment is supposed to be `no matching users were found`

File: src/js/components/dashboard/DashboardContent.jsx [hunk 3]
```diff
@@ -10,178 +10,199 @@ import DashboardFilters from './DashboardFilters';
...
-    certifiedSubmissions: PropTypes.array,
-    type: PropTypes.string,
-    activeTotal: PropTypes.number,
-    certifiedTotal: PropTypes.number,
-    activeLoading: PropTypes.bool,
-    certifiedLoading: PropTypes.bool,
-    toggleDashboardFilter: PropTypes.func,
-    stagedFilters: PropTypes.object,
-    appliedFilters: PropTypes.object
+  loadTableData: PropTypes.func,
```
Comment (ebdabbs) [inline src/js/components/dashboard/DashboardContent.jsx]: Some of these changes look like they are just differences in you IDE configuration (e.g. number of spaces indented). See https://github.com/fedspendingtransparency/data-act-documentation/blob/master/frontend_apps/STYLE_GUIDE.md#ide-configuration for style guide and linting plugins

File: src/js/components/dashboard/DashboardContent.jsx [hunk 4]
```diff
@@ -10,178 +10,199 @@ import DashboardFilters from './DashboardFilters';
...
-        );
-    }
+            <DashboardFilters
+              toggleFilter={this.toggleFilter}
+              bubbledRemovedFilterValue={this.state.bubbledRemovedFilterValue}
+              stagedFilters={stagedFilters.active}
+              appliedFilters={appliedFilters.active}
+              type={this.props.type}
+              table="active"
+            />
```
Comment (ebdabbs) [inline src/js/components/dashboard/DashboardContent.jsx]: Another minor code style difference - could you put `/>` on the same line as the last prop (with a space between) for consistency?

File: src/js/containers/dashboard/CreatedByContainer.jsx [hunk 5]
```diff
@@ -0,0 +1,142 @@
+/**
+ * CreatedByContainer.jsx
+ * Created by Lizzie Salita 8/30/18
```
Comment (ebdabbs) [inline src/js/containers/dashboard/CreatedByContainer.jsx]: Update file heading

File: src/js/components/dashboard/DashboardContent.jsx (hunk 3 above)
Comment (kojodoesdesign) [reply src/js/components/dashboard/DashboardContent.jsx]: IDE Options Updated!

**Pull request merged**